src/libs/assay_type.py

```diff
@@ -118,14 +126,32 @@ def main() -> None:
     """
     Some test routines
     """
-    for name, note in [('codex', 'this should fail'),
-                       ('CODEX', 'this should work'),
-                       ('codex_cytokit', 'this is not primary'),
-                       ('salmon_rnaseq_bulk', 'this is an alt name'),
-                       (['PAS', 'Image Pyramid'],
-                        'this is a complex alt name'),
-                       (['IMC', 'foo'],
-                        'this is an invalid complex alt name')]:
+    cases = [('codex',
```

Comment:
This testing approach is getting awkward. It essentially splits the assertions between two files: this one, and the output that you diff. Problems noticed in the diff would need to be found here and corrected. Suggest using doctests or pytest.

Reply:
I've converted it to doctest.  I'd thought that doing so would produce something that was unacceptably verbose, but it's not really that bad, and as you say the unification of the test prototype and the code itself has merit.